monkey/infection_monkey/utils/aws_environment_check.py

```diff
@@ -0,0 +1,22 @@
+import logging
+
+from common.cloud.aws.aws_instance import AwsInstance
+from infection_monkey.telemetry.aws_instance_telem import AwsInstanceTelemetry
+
+logger = logging.getLogger(__name__)
+
+
+def _running_on_aws(aws_instance: AwsInstance) -> bool:
+    return aws_instance.is_instance()
+
+
+def report_aws_environment():
+    logger.info("Collecting AWS info")
+
+    aws_instance = AwsInstance()
+
+    if _running_on_aws(aws_instance):
+        logger.info("Machine is an AWS instance")
+        AwsInstanceTelemetry({"instance_id": aws_instance.get_instance_id()}).send()
+    else:
+        logger.info("Machine is NOT an AWS instance")
```

Comment:
How long does this take to run if we're not running on AWS? Are there reasonable timeouts set?

Should we run this in a separate thread, so that it doesn't block execution of the agent while it's waiting for a timeout?

Reply:
`AwsInstance` has timeout of 2. It is making requests in order to get the instance id and the account id. For the region it doesn't have any timeout ( maybe we can set it up to be the same ) .

It is blocking the execution of the monkey so I will run it in a separate thread. Thanks!